src/client/pages/Offer/ProductSelector/Selector.tsx

```diff
@@ -32,34 +83,17 @@ export const Selector = ({
   const { search: searchParams } = useLocation()
   const history = useHistory()
 
-  const selectedProductIds = new URLSearchParams(searchParams).getAll(
-    SEARCH_PARAM,
-  )
-  const selectedStandaloneProducts = standaloneProducts.filter(({ id }) =>
-    selectedProductIds.includes(id),
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const [selectedProductsByCategory, setSelectedProductsByCategory] = useState(
+    getStateFromURL({ standaloneProducts, additionalProducts, searchParams }),
```

Comment:
Can we watch URL changes instead of local state?

Reply:
hmm yeah I don't think this PR contains all the changes the @guilhermespopolin and I made 🤔